kqueen/engines/openstack_kubespray.py

```diff
@@ -807,7 +813,40 @@ def _get_userdata(self):
         }
         return "#cloud-config\n" + yaml.dump(userdata)
 
-    def _boot_servers(self, *, name, servers_range, image, flavor, network,
+    def _set_up_security_groups(self):
+        master_sg = self.c.get_security_group(MASTER_SECURITY_GR)
+        if not master_sg:
+            master_sg = self.c.create_security_group(name=MASTER_SECURITY_GR,
```

Comment:
imo we need a try/catch here

Reply:
error catching is made in the above level (_run_provisioning function)